src/plugins/irc-events/link.js

```diff
@@ -23,14 +23,19 @@ module.exports = function(client, chan, msg) {
 		return;
 	}
 
-	const link = escapeHeader(links[0].link);
-	fetch(link, function(res) {
-		if (res === null) {
-			return;
-		}
+	Array.from(new Set( // Remove duplicate links
```

Comment:
Could you add proper caching instead of just deduping? As long as lounge process is running, it would keep X recent urls in memory (and block while one request is in flight, like await an existing promise).